src/routes/transactions/mappers/common/transactions.mapper.ts

```diff
@@ -0,0 +1,182 @@
+    | ModuleTransaction[]
+    | EthereumTransaction[];
+}
+
+@Injectable()
+export class TransactionsMapper {
+  constructor(
+    private readonly multisigTransactionMapper: MultisigTransactionMapper,
+    private readonly moduleTransactionMapper: ModuleTransactionMapper,
+    private readonly incomingTransferMapper: IncomingTransferMapper,
+  ) {}
+  async mapTransactions(
+    chainId: string,
+    transactionsDomain: TransactionDomain[],
+    safe: Safe,
+    offset: number,
+    timezoneOffset?: string,
+  ): Promise<[TransactionItem | DateLabel]> {
+    const prevPageTimestamp = this.getPreviousDayTimestamp(
+      transactionsDomain,
+      offset,
+      timezoneOffset,
+    )?.getTime();
+    // Remove first item that was requested to get previous day timestamp
+    if (prevPageTimestamp !== undefined) {
+      transactionsDomain = transactionsDomain.slice(1);
+    }
+    const transactionsDomainGroups = this.groupByDay(
+      transactionsDomain,
+      timezoneOffset,
+    );
+
+    const transactionList = await Promise.all(
+      transactionsDomainGroups.map(async (transactionGroup) => {
+        const transactions: (TransactionItem | DateLabel)[] = [];
+        if (transactionGroup.timestamp != prevPageTimestamp) {
+          transactions.push(new DateLabel(transactionGroup.timestamp));
+        }
+        transactions.push(
+          ...(await this.mapGroupTransactions(transactionGroup, chainId, safe)),
+        );
+        return await Promise.all(transactions.flat());
+      }),
+    );
+
+    return <[TransactionItem | DateLabel]>transactionList.flat();
+  }
+
+  private getPreviousDayTimestamp(
+    transactions: TransactionDomain[],
+    offset: number,
+    timezoneOffset?: string,
+  ): Date | undefined {
+    if (offset > 0) {
+      // Get previous page label
+      const timestamp =
+        transactions[0].executionDate ??
+        (transactions[0] as MultisigTransaction).submissionDate;
+      if (timestamp !== null) {
+        return this.getDayDate(timestamp, timezoneOffset);
+      } else {
+        throw Error('ExecutionDate cannot be null');
+      }
+    }
+  }
+
+  private groupByDay(
+    transactions: TransactionDomain[],
+    timezoneOffset?: string,
+  ): TransactionDomainGroup[] {
+    return Object.entries(
+      groupBy(transactions, (transaction) => {
+        let transactionTimestamp;
+        if (isMultisigTransaction(transaction)) {
+          transactionTimestamp =
+            transaction.executionDate ?? transaction.submissionDate;
+        } else {
+          transactionTimestamp = transaction.executionDate;
+        }
+        return this.getDayDate(transactionTimestamp, timezoneOffset).getTime();
+      }),
+    ).map(
+      ([timestamp, transactions]) =>
+        <TransactionDomainGroup>{
+          timestamp: Number(timestamp),
+          transactions: transactions,
+        },
+    );
+  }
+
+  private getDayDate(timestamp: Date, timezoneOffset?: string): Date {
```

Comment:
nitpick: I think `getDayFromDate` represents better what the function does.

Reply:
Done 0dac6dc15d88b563a60ab7a8d7d8e6367360f35f